orangecontrib/text/corpus.py

```diff
@@ -100,6 +100,30 @@ def extend_corpus(self, metadata, Y):
         self.X = self.W = np.zeros((len(self), 0))
         Table._init_ids(self)
 
+    def update_attributes(self, X, dictionary):
+        """
+        Adds X to the corpus and updates its domain accordingly.
+        """
+        self.X = X
+
+        features = [
+            dictionary[key]
+            for key
+            in range(len(dictionary.keys()))
+        ]
+        attr_value_set = [val for val in list(set(X.flatten()))]
+        attr = [
+            DiscreteVariable.make(f, attr_value_set)
```

Comment:
Also, since this is `DiscreteVariable` it is currently super slow, even for `bookexcerpts`.